client/src/lib/auphonic.api.ts

```diff
@@ -0,0 +1,107 @@
+import { curry } from 'lodash'
+import { addQuery, responseParser, ApiOptions } from './api'
+
+const defaultHeaders = (
+  { bearer }: { bearer?: string },
+  headers: { [key: string]: string } = {}
+) => ({
+  'Content-Type': 'application/json',
+  Accept: 'application/json',
+  ...authHeaders({ bearer }),
+  ...headers,
+})
+
+const authHeaders = ({ bearer }: { bearer?: string }) => ({
+  ...(bearer ? { Authorization: `Bearer ${bearer}` } : {}),
+})
+
+const readApi =
+  ({
+    errorHandler,
+    bearer,
+    method,
+    urlProcessor,
+  }: {
+    errorHandler?: Function
+    bearer?: string
+    method: 'GET' | 'DELETE'
+    urlProcessor?: (url: string) => string
+  }) =>
+  (url: string, { headers, query }: ApiOptions = {}) =>
+    fetch(addQuery(urlProcessor ? urlProcessor(url) : url, query), {
+      method,
+      headers: defaultHeaders({ bearer }, headers),
+    }).then(responseParser(errorHandler))
+
+const createApi =
+  ({
+    errorHandler,
+    bearer,
+    method,
+    urlProcessor,
+  }: {
+    errorHandler?: Function
+    bearer?: string
+    method: 'POST' | 'PUT'
+    urlProcessor?: (url: string) => string
+  }) =>
+  (url: string, data: any, { headers, query }: ApiOptions = {}) => {
+    return fetch(addQuery(urlProcessor ? urlProcessor(url) : url, query), {
+      method,
+      headers: defaultHeaders({ bearer }, headers),
+      body: JSON.stringify(data),
+    }).then(responseParser(errorHandler))
+  }
+
+const uploadApi =
+  ({
+    errorHandler,
+    bearer,
+    urlProcessor,
+  }: {
+    errorHandler?: Function
+    bearer?: string
+    urlProcessor?: (url: string) => string
+  }) =>
+  (url: string, data: any, { query }: ApiOptions = {}) => {
+    // TODO: this works but without progress reporting, apparently not possible with fetch
```

Comment:
wir können auch sowas wie `axios` verwenden, wollte ich bisher nicht weil wir noch keinen konkreten Bedarf hatten, würde aber viel Boilerplate ersparen :)

Reply:
Hab's hinzugefügt, aber nur an dieser Stelle. Du möchtest da sicher von den globalen configs für z.B. auth Header Gebraucht machen. Aber das überlasse ich dir, auch weil ich nicht weiß, ob man VueAxios verwenden will oder nicht, wie/wo du die globalen Sachen hinpackst, und so 🤷‍♂️

Reply:
Ne einfach nur axios. Vue axios verleitet nur die requests in der componente zu starten.